python/lsst/daf/butler/exprParser/exprTree.py

```diff
@@ -201,6 +201,38 @@ def __str__(self):
         return "{name}".format(**vars(self))
 
 
+class RangeLiteral(Node):
+    """Node representing range literal appearing in `IN` list.
+
+    Range literal defines a range of integer numbers with start and
+    end of the range (with inclusive end) and optional stride value
+    (default is 1).
+
+    Attributes
+    ----------
+    start : `int`
+        Start value of a range.
+    stop : `int`
+        End value of a range, inclusive, same or higher than ``start``.
+    stride : `int` or `None`
```

Comment:
Indeed I need to be more consistent about stride=1 vs None. I want to keep None for case when stride was not specified just in case some consumers of the parsed expression want to know exact syntax of the original expression (for whatever reason). Substitution of None to 1 should happen on the consumer side.I'll update code for consistency and also add stride=0 check, that can be done at parsing stage, there is no useful case for stride=0.